ccl/logictestccl: TestCCLLogic failed

[(ccl/logictestccl).TestCCLLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2733777&tab=buildLog) on [master@9ba48738bc511ad6954682cab41e23b8492facd8](https://github.com/cockroachdb/cockroach/commits/9ba48738bc511ad6954682cab41e23b8492facd8):

```
=== RUN   TestCCLLogic
    test_log_scope.go:73: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic722860822
    test_log_scope.go:74: use -show-logs to present logs inline
=== CONT  TestCCLLogic
    logic.go:3351: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic722860822
--- FAIL: TestCCLLogic (46.18s)
=== RUN   TestCCLLogic/multiregion-9node-3region-3azs
    --- FAIL: TestCCLLogic/multiregion-9node-3region-3azs (0.00s)
=== RUN   TestCCLLogic/multiregion-9node-3region-3azs/multi_region
=== PAUSE TestCCLLogic/multiregion-9node-3region-3azs/multi_region
=== CONT  TestCCLLogic/multiregion-9node-3region-3azs/multi_region
--- done: testdata/logic_test/regional_by_row_rename_column with config multiregion-9node-3region-3azs: 8 tests, 0 failures
--- total progress: 636 statements/queries
=== CONT  TestCCLLogic/multiregion-9node-3region-3azs/multi_region
    logic.go:2300: 
         
        testdata/logic_test/multi_region:650: SHOW ZONE CONFIGURATION FOR DATABASE alter_survive_db
        expected:
            DATABASE alter_survive_db  ALTER DATABASE alter_survive_db CONFIGURE ZONE USING
                                       range_min_bytes = 134217728,
                                       range_max_bytes = 536870912,
                                       gc.ttlseconds = 90000,
                                       num_replicas = 5,
                                       num_voters = 3,
                                       constraints = '{+region=ap-southeast-2: 1, +region=ca-central-1: 1, +region=us-east-1: 1}',
                                       voter_constraints = '[+region=ca-central-1]',
                                       lease_preferences = '[[+region=ca-central-1]]'
            
        but found (query options: "") :
            DATABASE alter_survive_db  ALTER DATABASE alter_survive_db CONFIGURE ZONE USING
                                       range_min_bytes = 134217728,
                                       range_max_bytes = 536870912,
                                       gc.ttlseconds = 90000,
                                       num_replicas = 4,
                                       num_voters = 3,
                                       constraints = '{+region=ap-southeast-2: 1, +region=ca-central-1: 1}',
                                       voter_constraints = '[+region=ca-central-1]',
                                       lease_preferences = '[[+region=ca-central-1]]'
--- done: testdata/logic_test/multi_region with config multiregion-9node-3region-3azs: 108 tests, 1 failures
E210302 06:13:22.098145 93225 sql/logictest/logic.go:3583  [-] 15  testdata/logic_test/multi_region:663: too many errors encountered, skipping the rest of the input
    logic.go:3126: 
        testdata/logic_test/multi_region:663: error while processing
    logic.go:3126: testdata/logic_test/multi_region:663: too many errors encountered, skipping the rest of the input
        --- FAIL: TestCCLLogic/multiregion-9node-3region-3azs/multi_region (9.95s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestCCLLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestCCLLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/logictestccl).TestCCLLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2744372&tab=buildLog) on [master@37cdf01299861235888af3411e699480f5a2d670](https://github.com/cockroachdb/cockroach/commits/37cdf01299861235888af3411e699480f5a2d670):

```
    logic.go:3351: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic451726180
--- FAIL: TestCCLLogic (53.73s)
=== RUN   TestCCLLogic/multiregion-9node-3region-3azs/multi_region
=== PAUSE TestCCLLogic/multiregion-9node-3region-3azs/multi_region
=== CONT  TestCCLLogic/multiregion-9node-3region-3azs/multi_region
=== CONT  TestCCLLogic/multiregion-9node-3region-3azs/multi_region
    logic.go:2300: 
         
        testdata/logic_test/multi_region:1106: SHOW ZONE CONFIGURATION FOR DATABASE zone_config_drop_region
        expected:
            DATABASE zone_config_drop_region  ALTER DATABASE zone_config_drop_region CONFIGURE ZONE USING
                                      range_min_bytes = 134217728,
                                      range_max_bytes = 536870912,
                                      gc.ttlseconds = 90000,
                                      num_replicas = 4,
                                      num_voters = 3,
                                      constraints = '{+region=ap-southeast-2: 1, +region=ca-central-1: 1}',
                                      voter_constraints = '[+region=ca-central-1]',
                                      lease_preferences = '[[+region=ca-central-1]]'
        but found (query options: "") :
            DATABASE zone_config_drop_region  ALTER DATABASE zone_config_drop_region CONFIGURE ZONE USING
                                              range_min_bytes = 134217728,
                                              range_max_bytes = 536870912,
                                              gc.ttlseconds = 90000,
                                              num_replicas = 5,
                                              num_voters = 3,
                                              constraints = '{+region=ap-southeast-2: 1, +region=ca-central-1: 1, +region=us-east-1: 1}',
                                              voter_constraints = '[+region=ca-central-1]',
                                              lease_preferences = '[[+region=ca-central-1]]'
--- done: testdata/logic_test/multi_region with config multiregion-9node-3region-3azs: 183 tests, 1 failures
--- progress: testdata/logic_test/regional_by_row: 121 statements/queries
--- total progress: 837 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 128 statements/queries
--- progress: testdata/logic_test/regional_by_row: 140 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 147 statements/queries
--- progress: testdata/logic_test/regional_by_row: 154 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 159 statements/queries
--- done: testdata/logic_test/regional_by_row with config multiregion-9node-3region-3azs: 166 tests, 0 failures
--- total progress: 1003 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 171 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 185 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 199 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 209 statements/queries
--- done: testdata/logic_test/alter_table_locality with config multiregion-9node-3region-3azs: 223 tests, 0 failures
--- total progress: 1226 statements/queries
--- total: 1226 tests, 1 failures
        --- FAIL: TestCCLLogic/multiregion-9node-3region-3azs/multi_region (13.15s)
=== RUN   TestCCLLogic/multiregion-9node-3region-3azs
    --- FAIL: TestCCLLogic/multiregion-9node-3region-3azs (0.00s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestCCLLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestCCLLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/logictestccl).TestCCLLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2749097&tab=buildLog) on [master@f0c2d8a2f762ee33ca044ae86f46d06fc35682db](https://github.com/cockroachdb/cockroach/commits/f0c2d8a2f762ee33ca044ae86f46d06fc35682db):

```
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestCCLLogic596766398
--- FAIL: TestCCLLogic (49.38s)
=== RUN   TestCCLLogic/multiregion-9node-3region-3azs/multi_region
=== PAUSE TestCCLLogic/multiregion-9node-3region-3azs/multi_region
=== CONT  TestCCLLogic/multiregion-9node-3region-3azs/multi_region
=== CONT  TestCCLLogic/multiregion-9node-3region-3azs/multi_region
    logic.go:2300: 
         
        testdata/logic_test/multi_region:1233: SHOW ZONE CONFIGURATION FOR DATABASE zone_config_drop_region
        expected:
            DATABASE zone_config_drop_region  ALTER DATABASE zone_config_drop_region CONFIGURE ZONE USING
                                      range_min_bytes = 134217728,
                                      range_max_bytes = 536870912,
                                      gc.ttlseconds = 90000,
                                      num_replicas = 4,
                                      num_voters = 3,
                                      constraints = '{+region=ap-southeast-2: 1, +region=ca-central-1: 1}',
                                      voter_constraints = '[+region=ca-central-1]',
                                      lease_preferences = '[[+region=ca-central-1]]'
        but found (query options: "") :
            DATABASE zone_config_drop_region  ALTER DATABASE zone_config_drop_region CONFIGURE ZONE USING
                                              range_min_bytes = 134217728,
                                              range_max_bytes = 536870912,
                                              gc.ttlseconds = 90000,
                                              num_replicas = 5,
                                              num_voters = 3,
                                              constraints = '{+region=ap-southeast-2: 1, +region=ca-central-1: 1, +region=us-east-1: 1}',
                                              voter_constraints = '[+region=ca-central-1]',
                                              lease_preferences = '[[+region=ca-central-1]]'
--- done: testdata/logic_test/multi_region with config multiregion-9node-3region-3azs: 202 tests, 1 failures
--- total progress: 859 statements/queries
--- progress: testdata/logic_test/regional_by_row: 143 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 134 statements/queries
--- progress: testdata/logic_test/regional_by_row: 163 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 155 statements/queries
--- progress: testdata/logic_test/regional_by_row: 177 statements/queries
--- done: testdata/logic_test/regional_by_row with config multiregion-9node-3region-3azs: 177 tests, 0 failures
E210305 14:07:24.463472 125452 sql/stats/automatic_stats.go:302  [-] 26  failed to refresh stats: node unavailable; try another peer
--- total progress: 1036 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 171 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 187 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 199 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 216 statements/queries
--- done: testdata/logic_test/alter_table_locality with config multiregion-9node-3region-3azs: 225 tests, 0 failures
E210305 14:07:32.427991 124674 sql/stats/automatic_stats.go:302  [-] 27  failed to refresh stats: node unavailable; try another peer
--- total progress: 1261 statements/queries
--- total: 1261 tests, 1 failures
        --- FAIL: TestCCLLogic/multiregion-9node-3region-3azs/multi_region (14.03s)
=== RUN   TestCCLLogic/multiregion-9node-3region-3azs
    --- FAIL: TestCCLLogic/multiregion-9node-3region-3azs (0.00s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestCCLLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestCCLLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(ccl/logictestccl).TestCCLLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2749254&tab=buildLog) on [master@f0c2d8a2f762ee33ca044ae86f46d06fc35682db](https://github.com/cockroachdb/cockroach/commits/f0c2d8a2f762ee33ca044ae86f46d06fc35682db):

```
            DATABASE alter_survive_db  ALTER DATABASE alter_survive_db CONFIGURE ZONE USING
                                       range_min_bytes = 134217728,
                                       range_max_bytes = 536870912,
                                       gc.ttlseconds = 90000,
                                       num_replicas = 5,
                                       num_voters = 3,
                                       constraints = '{+region=ap-southeast-2: 1, +region=ca-central-1: 1, +region=us-east-1: 1}',
                                       voter_constraints = '[+region=ca-central-1]',
                                       lease_preferences = '[[+region=ca-central-1]]'
            
        but found (query options: "") :
            DATABASE alter_survive_db  ALTER DATABASE alter_survive_db CONFIGURE ZONE USING
                                       range_min_bytes = 134217728,
                                       range_max_bytes = 536870912,
                                       gc.ttlseconds = 90000,
                                       num_replicas = 4,
                                       num_voters = 3,
                                       constraints = '{+region=ap-southeast-2: 1, +region=ca-central-1: 1}',
                                       voter_constraints = '[+region=ca-central-1]',
                                       lease_preferences = '[[+region=ca-central-1]]'
--- done: testdata/logic_test/multi_region with config multiregion-9node-3region-3azs: 123 tests, 1 failures
E210305 15:25:36.373014 98662 sql/logictest/logic.go:3583  [-] 21  testdata/logic_test/multi_region:765: too many errors encountered, skipping the rest of the input
    logic.go:3126: 
        testdata/logic_test/multi_region:765: error while processing
    logic.go:3126: testdata/logic_test/multi_region:765: too many errors encountered, skipping the rest of the input
E210305 15:25:36.933197 163026 jobs/adopt.go:260  [n1] 22  job 638607479539957761: adoption completed with error failed to ingest index entries during backfill: duplicate key value violates unique constraint "primary"
E210305 15:25:36.939462 164677 jobs/adopt.go:118  [-] 23  could not run claimed job 638607479539957761: job 638607479539957761: status changed to failed which is not resumable`
--- progress: testdata/logic_test/regional_by_row: 77 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 69 statements/queries
--- progress: testdata/logic_test/regional_by_row: 114 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 103 statements/queries
--- progress: testdata/logic_test/regional_by_row: 135 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 125 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 144 statements/queries
--- progress: testdata/logic_test/regional_by_row: 153 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 156 statements/queries
--- progress: testdata/logic_test/regional_by_row: 167 statements/queries
--- done: testdata/logic_test/regional_by_row with config multiregion-9node-3region-3azs: 177 tests, 0 failures
--- total progress: 834 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 167 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 183 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 195 statements/queries
--- progress: testdata/logic_test/alter_table_locality: 207 statements/queries
--- done: testdata/logic_test/alter_table_locality with config multiregion-9node-3region-3azs: 225 tests, 0 failures
E210305 15:25:56.033539 207169 kv/kvclient/kvcoord/txn_interceptor_committer.go:452  [-] 24  making txn commit explicit failed for "unnamed" meta={id=89bfbc15 pri=0.00471910 epo=0 ts=1614957956.007899752,0 min=1614957956.007899752,0 seq=2} lock=true stat=STAGING rts=1614957956.007899752,0 wto=false gul=1614957956.507899752,0 ifw=1: failed to send RPC: sending to all replicas failed; last error: node unavailable; try another peer
--- total progress: 1059 statements/queries
--- total: 1059 tests, 0 failures
        --- FAIL: TestCCLLogic/multiregion-9node-3region-3azs/multi_region (7.49s)
=== RUN   TestCCLLogic/multiregion-9node-3region-3azs
    --- FAIL: TestCCLLogic/multiregion-9node-3region-3azs (0.01s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestCCLLogic PKG=./pkg/ccl/logictestccl TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestCCLLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/type_change.go
+++ b/pkg/sql/type_change.go
@@ -412,7 +412,10 @@ func performMultiRegionFinalization(
 	descsCol *descs.Collection,
 ) ([]descpb.ID, error) {
 	_, dbDesc, err := descsCol.GetImmutableDatabaseByID(
-		ctx, txn, typeDesc.ParentID, tree.DatabaseLookupFlags{Required: true})
+		ctx, txn, typeDesc.ParentID, tree.DatabaseLookupFlags{
+			AvoidCached: true,
+			Required:    true,
+		})
 	if err != nil {
 		return nil, err
 	}

